envisage/provider_extension_registry.py

```diff
@@ -103,10 +103,7 @@ def _get_extensions(self, extension_point_id):
         #     all = [x for y in extensions for x in y]
         #
         # But I'm sure sure that that makes it any clearer ;^)
-        all = []
-        map(all.extend, extensions)
-
-        return all
+        return [x for y in extensions for x in y]
```

Comment:
Please use better names than `x` and `y`, which will help with readability and understanding of the list comprehension.